src/compiler/checker.ts

```diff
@@ -6235,6 +6235,9 @@ namespace ts {
             if (relation === assignableRelation || relation === comparableRelation) {
                 if (source.flags & TypeFlags.Any) return true;
                 if (source.flags & (TypeFlags.Number | TypeFlags.NumberLiteral) && target.flags & TypeFlags.Enum) return true;
+                if (source.flags & (TypeFlags.Number | TypeFlags.NumberLiteral) &&
+                    target.flags & TypeFlags.Union &&
+                    forEach((target as UnionType).types, t => t.flags & TypeFlags.EnumLike)) return true;

```

Comment:
Actually, this code shouldn't be checking for unions at all. The outer `isRelatedTo` machinery handles it. The much simpler fix is to just change the previous line from `TypeFlags.Enum` to `TypeFlags.EnumLike`.